src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -164,6 +159,36 @@ ITK_coordinates_to_STIR_physical_coordinates
   return stir_coord;
 }
 
+/* Convert an ITK Pixel (i.e., float) to a STIR Pixel. */
+typename STIRImageSingle::pixel_type
```

Comment:
are you sure you need the `typename`? Should normally only be in a template (also below of course).

Could `inline` this. should make it `static`

Reply:
A lot of these are functions, not members. I was just following from what was previously done in this file. So does static make sense?

Reply:
As for the typename thing, I've now changed it back to how I had originally planned. First function is templated, second is a specialisation. You still need to explicitly define the types when used so that the return type can be inferred.

Reply:
I made all the new functions static and inline, but I am curious what static does for non-member functions? I didn't think it would compile. A quick Google didn't show much.

Reply:
Everywhere inline and static (not static on specialisations, compile error). What exactly does static do for non-member functions? I had a Google, but couldn't find any info.

Reply:
> What exactly does static do for non-member functions?

It makes them local to the translation unit, i.e. no chance anyone can ever link to them (otherwise you can, even if there's no .h for them)

Reply:
Good to know :)